lib/que/metrics.rb

```diff
@@ -80,6 +50,22 @@ def expose(port: 8080)
       )
     end
 
+    def initialize(labels: {}, registry: Prometheus::Client.registry)
+      @registry = self.class.register_metrics(registry)
+      @metrics = @registry.metrics.each_with_object({}) { |m, ms| ms[m.name] = m }
+      @base_labels = labels
+
+      # These instance variables all relate to internal running job metrics. They are used
+      # to periodically update the jobs_worked_seconds counter.
+      @running_job_lock = Mutex.new
+      @running_job_labels = nil
+      @running_job_last_seen = nil
+      @running_job_tracker_stop = start_job_tracker
+
+      # When we get garbage collected, stop the job tracker thread.
```

Comment:
With threaded workers, we may be constructing many workers and could easily forget to stop a worker, even if it's been garbage collected. We can improve usability by registering an object finalizer that tells the runtime tracker to stop, preventing what may have otherwise been a resource leak/inaccurate metric counter.